src/Interpreters/executeQuery.cpp

```diff
@@ -1009,8 +1050,15 @@ static std::tuple<ASTPtr, BlockIO> executeQueryImpl(
     }
     catch (...)
     {
-        if (auto txn = context->getCurrentTransaction())
+        if (implicit_txn_control)
+        {
+            implicit_txn_control->executeRollback(context->getSessionContext());
```

Comment:
Hm, seems like it may unexpectedly throw "There is no current transaction" if `wait_changes_become_visible_after_commit_mode` is not `WAIT_UNKNOWN` and ZK connection was lost during commit:
https://github.com/ClickHouse/ClickHouse/blob/e4d4d0a8543700c2bc85b688c9f8f6cfff64cc48/src/Interpreters/InterpreterTransactionControlQuery.cpp#L67-L71

Reply:
Yep, you are right. I'll wrap the commit call with a try catch to make sure `implicit_txn_control->reset()` is called and avoid trying to rollback later